### DISCUSSION TITLE
ACPI and Redfish

### DISCUSSION THREAD

**Discussion by marvin-steinke** [sustainable-computing-io/kepler#1662] [Q&A] [answered]
Some of your diagrams for Bare-Metal Deployment state that RAPL, NVML, ACPI and Redfish are used for energy readings.

How do ACPI and Redfish relate?
ACPI is a standard for device configuration and power management by the operating system. Redfish is a standard for managing and interfacing with hardware through a RESTful API.

Does Redfish use ACPI? Do they report different energy readings?

**Comment by aharivel** [comment]
You can retrieve energy metrics using Redfish ( check slide  n°7 https://www.dmtf.org/sites/default/files/SPMF%20Introduction%20to%20Redfish%20May%202016.pdf ).
With ACPI, it depends on the architecture and the systems itself, here a pointer for you:  https://www.kernel.org/doc/Documentation/hwmon/acpi_power_meter 

Hope it helps
Anthony

**Comment by marvin-steinke** [comment, marked as answer]
I took a look at the implementation and found out that Redfish is favored over ACPI if it is available and they are not used in conjunction. ACPI seems to be more about the system's power state and configuration, mainboard voltage and thermal sensors, etc., while many modern hardware components report their power consumption to the Redfish API which can then be queried.
